spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/web/reactive/WebFluxAutoConfiguration.java

```diff
@@ -149,14 +149,16 @@ public void addResourceHandlers(ResourceHandlerRegistry registry) {
 				return;
 			}
 			Duration cachePeriod = this.resourceProperties.getCache().getPeriod();
+			CacheControl cacheControl = this.resourceProperties.getCache()
+					.getCachecontrol().toHttpCacheControl();
 			if (!registry.hasMappingForPattern("/webjars/**")) {
 				ResourceHandlerRegistration registration = registry
 						.addResourceHandler("/webjars/**")
 						.addResourceLocations("classpath:/META-INF/resources/webjars/");
 				if (cachePeriod != null) {
-					registration.setCacheControl(CacheControl
-							.maxAge(cachePeriod.toMillis(), TimeUnit.MILLISECONDS));
+					cacheControl.sMaxAge(cachePeriod.toMillis(), TimeUnit.MILLISECONDS);
```

Comment:
`sMaxAge` looks like a typo here - if the current tests are not breaking with that, they're probably not testing it right.